install

```diff
@@ -1,29 +1,32 @@
+#!/usr/bin/env sh
+
+set -eu
+
 get_latest_release() {
-	curl --silent "https://api.github.com/repos/$repo/releases/latest" | # Get latest release from GitHub api
-		grep '"tag_name":' |                                            # Get tag line
-		sed -E 's/.*"([^"]+)".*/\1/'                                    # Pluck JSON value
-	}
+  repo="$1"
+	curl -sSL "https://api.github.com/repos/${repo}/releases/latest" | \
+    awk 'BEGIN{FS=": |,|\""}; /tag_name/{print $5}'
+}
 
-repo='huacnlee/gobackup'
-version=`get_latest_release`
-if [[ `uname` == 'Darwin' ]]; then
-   platform='darwin'
-else
-   platform='linux'
-fi
-if [[ `arch` == 'arm64' ]]; then
-	arch='arm64'
-else
-	arch='amd64'
-fi
-curl -SLo gobackup.tar.gz https://github.com/huacnlee/gobackup/releases/download/$version/gobackup-$platform-$arch.tar.gz
-tar zxf gobackup.tar.gz
+repo="huacnlee/gobackup"
+version="$(get_latest_release "${repo}")"  # v1.2.0
+platform="$(uname | tr "[A-Z]" "[a-z]")"  # Linux => linux
+arch="$(uname -m | sed 's/x86_64/amd64/')"  # x86_64 => amd64
+package="gobackup-${platform}-${arch}.tar.gz"
+package_url="https://github.com/${repo}/releases/download/${version}/${package}"
+bin="gobackup"
+dest_dir="/usr/local/bin"
+tmp_dir="$(mktemp -d)"
+
+trap "rm -r ${tmp_dir}" EXIT
+
+cd "${tmp_dir}"
+curl -sSL "${package_url}" | tar xzf -
 
-if [[ `whoami` == 'root' ]]; then
-   mv gobackup /usr/local/bin/gobackup
+if test $(id -u) -eq 0; then
+   mv "${bin}" "${dest_dir}"
 else
-   sudo mv gobackup /usr/local/bin/gobackup
+   sudo mv "${bin}" "${dest_dir}"
 fi
-mkdir -p ~/.gobackup && touch ~/.gobackup/gobackup.yml
-rm gobackup.tar.gz
 
+mkdir -p ~/.gobackup
```

Comment:
touch ~/.gobackup/gobackup.yml

Reply:
That's what I debate in the comment.

IMO an empty `~/.gobackup/gobackup.yml` only makes `gobackup perform` doesn't has config not found error, not much else, meanwhile it may makes new user more confusing:
1. User doesn't know there's an empty `~/.gobackup/gobackup.yml`.
2. `/etc/gobackup/gobackup.yml` won't work since `~/.gobackup/gobackup.yml` has higher priority. `gobackup perform` won't do anything in such scenario.
3. An empty `~/.gobackup/gobackup.yml` doesn't help user. If user wants to configure `~/.gobackup/gobackup.yml` then user'll create the file by self.
4. It's better to just let user know config should be configured.